pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -76,14 +78,31 @@ type Manager struct {
 }
 
 // NewManager returns a new node shutdown manager.
-func NewManager(getPodsFunc eviction.ActivePodsFunc, killPodFunc eviction.KillPodFunc, shutdownGracePeriodRequested, shutdownGracePeriodCriticalPods time.Duration) *Manager {
-	return &Manager{
+func NewManager(getPodsFunc eviction.ActivePodsFunc, killPodFunc eviction.KillPodFunc, syncNodeStatus func(), shutdownGracePeriodRequested, shutdownGracePeriodCriticalPods time.Duration) (*Manager, lifecycle.PodAdmitHandler) {
+	manager := &Manager{
 		getPods:                         getPodsFunc,
 		killPod:                         killPodFunc,
+		syncNodeStatus:                  syncNodeStatus,
 		shutdownGracePeriodRequested:    shutdownGracePeriodRequested,
 		shutdownGracePeriodCriticalPods: shutdownGracePeriodCriticalPods,
 		clock:                           clock.RealClock{},
 	}
+	return manager, manager
+}
+
+// Admit rejects all pods if node is shutting
+func (m *Manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
```

Comment:
can we please add a unit test that expires this functionality and tests:

1. pods are not admitted when `nodeShuttingDownNow == true`
2. pods are admitted when `nodeShuttingDownNow == false`